sql-streaming-copier/README.md

```diff
@@ -0,0 +1,127 @@
+    payload BYTES,
+)
+```
+
+If copying to Pub/Sub, all attribute values must be interpretable as UTF-8
+strings. If multiple values are provided for an attribute, they will be joined
+with commas in the Pub/Sub Message.
+
+If copying to BigQuery, the table must accept the above format if it already
+exists.
+
+It requires the following options:
+
+* sourceType: [`pubsub`, `pubsublite`, `kafka`]
+* sourceLocation:
+  * `pubsub`: The topic (discouraged) or subscription to read from
+  * `pubsublite`: The subscription to read from
+  * `kafka`: The broker URL to read from
+* sinkType: [`pubsub`, `pubsublite`, `kafka`, `bigquery`]
+* sinkLocation:
+  * `pubsub`: The topic to write to
+  * `pubsublite`: The topic to write to
+  * `kafka`: The broker URL to write to
+  * `bigquery`: Location of the table in the BigQuery CLI format
+  
+When running with kafka, you must also set the following parameters:
+
+```bash
+--parameters='[source/sink]Options=^NOT_IN_JSON^{"topics":["<kafka topic name>"],"bootstrap.servers":"<broker URL>"}'
+```
+
+Additional options from
+[beam documentation](https://beam.apache.org/documentation/dsls/sql/extensions/create-external-table)
+can be provided using the `sourceOptions` and `sinkOptions` parameters.
+
+## How to build
+
+### Shaded jar generation
+
+To regenerate the shaded jar, put the following into a
+`custom-shadowjar/build.gradle` package in the beam source tree, add
+`include(":custom-shadowjar")` to `settings.gradle.kts` and run
+`./gradlew :custom-shadowjar:shadowJar`.
+
+```groovy
+plugins {
+    id 'java'
+    id 'org.apache.beam.module'
+    id 'com.github.johnrengelman.shadow'
+}
+applyJavaNature(
+        automaticModuleName: 'org.apache.beam.sql-shaded'
+)
+
+dependencies {
+    implementation project(":sdks:java:extensions:sql")
+    implementation project(":sdks:java:extensions:sql:zetasql")
+    implementation project(":sdks:java:io:google-cloud-platform")
+    implementation project(":runners:direct-java")
+    implementation project(":runners:google-cloud-dataflow-java")
+    implementation project(":sdks:java:io:kafka")
+}
+
+shadowJar {
+    mergeServiceFiles()
+    zip64 true
+}
+```
+
+Then, run the following command:
+
+```bash
+export BUILT_JAR_PATH="<path to full beam jar>"
+export THIRD_PARTY_PATH="<path to git repo>/sql-streaming-copier/third_party"
+
+mvn org.apache.maven.plugins:maven-install-plugin:2.3.1:install-file
+-Dfile="$BUILT_JAR_PATH" 
+-DgroupId=com.google.cloud.pubsub
+-DartifactId=beam-custom-shadowjar
+-Dversion=1.0.0
+-Dpackaging=jar 
+-DlocalRepositoryPath="$THIRD_PARTY_PATH"
+```
+
+### Deploying
+
+First, from the `sql-streaming-copier` directory, run `mvn package`.
+
+Then, run the following to upload the new version of the template.
+
+```bash
+export TEMPLATE_PATH="gs://<TEMPLATE JSON FILE NAME>.json"
+export TEMPLATE_IMAGE="gcr.io/<PROJECT>/<CONTAINER PATH>:latest"
+
+gcloud dataflow flex-template build $TEMPLATE_PATH \
```

Comment:
I suspect we are going to update this to include a command to run using the publicly exposed template, right? In that way, people won't have to build this themselves.

Reply:
Thats not what this does. This is in the building section: it is instructions for how to deploy the template not run it. I've added an example gcloud command for running the template.